src/web-map.js

```diff
@@ -275,6 +276,36 @@ export class WebMap extends HTMLMapElement {
         this.dispatchEvent(new CustomEvent('createmap'));
       }
     }
+
+    /*
+      1. only deletes aria-label when the last (only remaining) map caption is removed
+      2. only deletes aria-label if the aria-label was defined by the map caption element itself
+    */
+
+    let mapcaption = this.querySelector('map-caption');
+    
+    if (mapcaption != null){
+      setTimeout(() => {
+        let ariaupdate = this.getAttribute('aria-label');
+  
+        if (ariaupdate == mapcaption.innerHTML) {
+          this.mapCaptionObserver = new MutationObserver((m) => {
+            let mapcaptionupdate = this.querySelector('map-caption');
+            if (mapcaptionupdate != mapcaption)
+            {
+              this.removeAttribute('aria-label');
+            }
+            
+          });
+          this.mapCaptionObserver.observe(this, {
+            characterData: true,
+            subtree: true,
+            attributes: true,
```

Comment:
Probably only need `attributes: true` here, no?  This is a big element with lots of subtree changes, could be a large performance burden due to all this observing going on.